src/components/TimePicker/__test__/timePicker.spec.js

```diff
@@ -1,34 +1,139 @@
+        const component = mount(
+            <TimePicker label="unit-testing-timePicker" placeholder="testing-timePicker" />,
+        );
+        const instance = component.instance();
+        const focusedElementPlaceholder = document.activeElement.getAttribute('placeholder');
+
+        instance.focus();
+        expect(document.activeElement.getAttribute('placeholder')).toBe(focusedElementPlaceholder);
+        instance.blur();
+        expect(document.activeElement.getAttribute('placeholder')).toBeNull();
+    });
+    it('should set isOpen to true when enter key or space key is pressed while input is focused and readOnly is not passed', () => {
+        const values = [ENTER_KEY, SPACE_KEY];
+        values.forEach(value => {
+            const component = mount(<TimePicker label="unit-testing-timePicker" />);
+            const timePickerInput = component.find('input');
+            timePickerInput.simulate('focus');
+            timePickerInput.simulate('keyDown', { keyCode: value });
+            expect(component.find('Modal').prop('isOpen')).toBe(true);
+        });
+    });
+    it('should not set isOpen to true when enter key or space key is pressed while input is focused and readOnly is passed', () => {
+        const values = [ENTER_KEY, SPACE_KEY];
+        values.forEach(value => {
+            const component = mount(<TimePicker label="unit-testing-timePicker" readOnly />);
+            const timePickerInput = component.find('input');
+            timePickerInput.simulate('focus');
+            timePickerInput.simulate('keyDown', { keyCode: value });
+            expect(component.find('Modal').prop('isOpen')).toBe(false);
+        });
+    });
+    it('should call to focusHourInput when the modal is open', () => {
+        const component = mount(<TimePicker label="unit-testing-timePicker" />);
+        component.find('input').simulate('click');
+        component.update();
+        component.instance().fieldRef.current.timeSelectRef.current.focusHourInput = jest.fn();
+        setTimeout(() => {
+            expect(
+                component.instance().fieldRef.current.timeSelectRef.current.focusHourInput,
+            ).toHaveBeenCalledTimes(1);
+        }, 1000);
+    });
+    it('should call updateValue when value change dynamically', () => {
+        const component = mount(<TimePicker label="unit-testing-timePicker" value="22:59" />);
+        component.instance().fieldRef.current.updateValue = jest.fn();
+        component.setProps({ value: '23:01' });
+        component.update();
+        expect(component.instance().fieldRef.current.updateValue).toHaveBeenCalledTimes(1);
+    });
+    it('should set the right value when value change dynamically', () => {
+        const component = mount(<TimePicker label="unit-testing-timePicker" value="22:59" />);
+        component.setProps({ value: '23:01' });
+        component.update();
+        expect(component.find('input').prop('value')).toBe('11:01 PM');
+    });
```

Comment:
this both above are one test
remember that this.updateValue is internal and we should not test that this.updateValue was called, but that the input get the new value like this last test do